Sources/Interaction/Widgets/ImageCroppingRegionsRepresentation/index.js

```diff
@@ -0,0 +1,384 @@
+    publicAPI.setOpacity(model.opacity);
+
+    model.centerMapper.setInputData(model.centerRegionPolyData);
+    model.centerActor.setMapper(model.centerMapper);
+
+    // set picker
+    model.cursorPicker = vtkCellPicker.newInstance();
+  }
+
+  publicAPI.getActors = () => [model.actor, model.centerActor];
+  publicAPI.getNestedProps = () => publicAPI.getActors();
+
+  // Reorders a bounds array such that each (a,b) pairing is a
+  // (min,max) pairing.
+  function reorderBounds(bounds) {
+    for (let i = 0; i < 6; i += 2) {
+      if (bounds[i] > bounds[i + 1]) {
+        const tmp = bounds[i + 1];
+        bounds[i + 1] = bounds[i];
+        bounds[i] = tmp;
+      }
+    }
+  }
+
+  function updateCenterOpacity() {
+    const slicePos = model.slice;
+    const [
+      xPLower,
+      xPUpper,
+      yPLower,
+      yPUpper,
+      zPLower,
+      zPUpper,
+    ] = model.planePositions;
+
+    let centerInvisible = true;
+    switch (model.sliceOrientation) {
+      case Orientation.YZ:
+        centerInvisible = xPLower < slicePos && slicePos < xPUpper;
+        break;
+      case Orientation.XZ:
+        centerInvisible = yPLower < slicePos && slicePos < yPUpper;
+        break;
+      case Orientation.XY:
+        centerInvisible = zPLower < slicePos && slicePos < zPUpper;
+        break;
+      default:
+      // noop
+    }
+
+    model.centerActor
+      .getProperty()
+      .setOpacity(centerInvisible ? 0 : model.opacity);
+  }
+
+  publicAPI.placeWidget = (...bounds) => {
+    const boundsArray = [];
+
+    for (let i = 0; i < bounds.length; i++) {
+      boundsArray.push(bounds[i]);
+    }
+
+    if (boundsArray.length !== 6) {
+      return;
+    }
+
+    // make sure each bounds pairing is monotonic
+    reorderBounds(boundsArray);
+
+    const newBounds = [];
+    const center = [];
+    publicAPI.adjustBounds(boundsArray, newBounds, center);
+
+    for (let i = 0; i < 6; i++) {
+      model.initialBounds[i] = newBounds[i];
+    }
+
+    model.initialLength = Math.sqrt(
+      (newBounds[1] - newBounds[0]) * (newBounds[1] - newBounds[0]) +
+        (newBounds[3] - newBounds[2]) * (newBounds[3] - newBounds[2]) +
+        (newBounds[5] - newBounds[4]) * (newBounds[5] - newBounds[4])
+    );
+
+    // set plane positions
+    publicAPI.setPlanePositions(...model.initialBounds);
+  };
+
+  publicAPI.setSlice = (slice) => {
+    if (slice !== model.slice) {
+      model.slice = slice;
+      publicAPI.updateGeometry();
+    }
+  };
+
+  publicAPI.setSliceOrientation = (sliceOrientation) => {
+    if (sliceOrientation !== model.sliceOrientation) {
+      model.sliceOrientation = sliceOrientation;
+      publicAPI.updateGeometry();
+    }
+  };
+
+  publicAPI.setOpacity = (opacityValue) => {
+    const opacity = Math.max(0, Math.min(1, opacityValue));
+    model.opacity = opacity;
+    model.actor.getProperty().setOpacity(opacity);
+    updateCenterOpacity();
+  };
+
+  publicAPI.setEdgeColor = (...edgeColor) => {
```

Comment:
Should set methods call `publicAPI.modified()`?

Reply:
only if `model.opacity !== opacity`...

Reply:
fixed